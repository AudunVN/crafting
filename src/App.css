/* http://meyerweb.com/eric/tools/css/reset/
   v2.0 | 20110126
   License: none (public domain)
*/
a, abbr, acronym, address, applet, article, aside, audio, b, big, blockquote, body, canvas, caption, center, cite, code, dd, del, details, dfn, div, dl, dt, em, embed, fieldset, figcaption, figure, footer, form, h1, h2, h3, h4, h5, h6, header, hgroup, html, i, iframe, img, ins, kbd, label, legend, li, mark, menu, nav, object, ol, output, p, pre, q, ruby, s, samp, section, small, span, strike, strong, sub, summary, sup, table, tbody, td, tfoot, th, thead, time, tr, tt, u, ul, var, video {
    margin: 0;
    padding: 0;
    border: 0;
    font: inherit
}

article, aside, details, figcaption, figure, footer, header, hgroup, menu, nav, section {
    display: block
}

body {
    line-height: 1
}

ol, ul {
    list-style: none
}

blockquote, q {
    quotes: none
}

blockquote:after, blockquote:before, q:after, q:before {
    content: '';
    content: none
}

table {
    border-collapse: collapse;
    border-spacing: 0
}

@font-face {
    font-family: 'Minecraft';
    src: url(assets/Minecraft.woff);
}

body {
    background: #eee;
    font-family: Minecraft;
    padding: 10px;
    color: #404040;
}

.crafting-holder, .ingredients {
    display: flex;
    display: -webkit-flex; /* Safari 8 */
    flex-wrap: wrap;
    -webkit-flex-wrap: wrap; /* Safari 8 */
    padding: 10px;
    margin-top: 10px;
    background: #C6C6C6;
    border: 2px solid #373737;
    border-right-color: #FFF;
    border-bottom-color: #FFF;
}

.ingredients {
    max-height: 500px;
    overflow-y: scroll;
}

.ingredients::-webkit-scrollbar {
    width: auto;
}

.ingredients::-webkit-scrollbar-track {
    background-color: #8B8B8B;
}

.ingredients::-webkit-scrollbar-thumb {
    height: 10px;
    background-color: #C6C6C6;
    border: 2px solid #373737;
    border-right-color: #FFF;
    border-bottom-color: #FFF;
}

.crafting {
    max-width: 230px;
    margin: 0 auto;
}

.crafting-table {
    background: #C6C6C6;
    text-align: left !important;
    margin-left: 2px;
    line-height: 42px;
}

.grid, .grid-large {
    float: left;
    display: block;
    text-align: center !important;
    background-color: #8B8B8B;
    border: 2px solid #373737;
    border-right-color: #FFF;
    border-bottom-color: #FFF;
    vertical-align: middle;
}

.grid-large img {
    z-index: -1;
    margin-top: 8px;
}

.grid {
    width: 36px;
    height: 36px;
}

.grid:hover, .grid-large:hover {
    background-color: rgba(255, 255, 255, 0.4);
}

.grid-large {
    width: 48px;
    height: 48px;
}

.faker {
}

.crafting .recipe {
    float: left;
    width: 110px;
}

.crafting .arrow {
    width: 32px;
    height: 27px;
    margin: 50px 15px 0 15px;
    float: left;
    background: url(assets/arrow.png);
}

.crafting-table-output {
    float: left;
    padding-top: 40px;
}

input[type="text"] {
    font-family: Minecraft;
    font-size: 1em;
    border: 2px solid #404040;
    background: #8B8B8B;
    padding: 2px;
    color: #FFFFFF;
}

.search-box {
    display: inline-flex;
    flex: 0 0 100%;
    margin-bottom: 5px;
}

.search-box input {
    flex: 2;
    margin-left: 10px;
}

.mc-tooltip {
    text-align: left;
    position: fixed;
    top: 0;
    left: 0;
    background-color: #100010;
    background-color: rgba(16, 0, 16, 0.94);
    color: #FFF;
    padding: 0.375em;
    font-family: Minecraft;
    font-size: 16px;
    word-spacing: 4px;
    white-space: nowrap;
    line-height: 1.25em;
    text-shadow: 0.125em 0.125em 0 #3F3F3F;
    margin: 0.125em 0.25em;
    pointer-events: none;
    z-index: 99;
}

.mc-tooltip:before {
    content: "";
    position: absolute;
    top: 0.125em;
    right: -0.125em;
    bottom: 0.125em;
    left: -0.125em;
    border: 0.125em solid #100010;
    border-style: none solid;
    border-color: rgba(16, 0, 16, 0.94);
}

.mc-tooltip:after {
    content: "";
    position: absolute;
    top: 0.125em;
    right: 0;
    bottom: 0.125em;
    left: 0;
    border: 0.125em solid #2D0A63;
    border-image: -webkit-linear-gradient(rgba(80, 0, 255, 0.31), rgba(40, 0, 127, 0.31)) 1;
    border-image: linear-gradient(rgba(80, 0, 255, 0.31), rgba(40, 0, 127, 0.31)) 1;
}

.mc-tooltip > .mc-tooltip-description {
    font-size: 12px;
    display: block;
    margin-top: 0.25em
}

@-webkit-keyframes shake {
    0% { -webkit-transform:     rotate(0deg); }
    10% { -webkit-transform:    rotate(-5deg); }
    20% { -webkit-transform:    rotate(5deg); }
    30% { -webkit-transform:    rotate(0deg); }
    40% { -webkit-transform:    rotate(5deg); }
    50% { -webkit-transform:    rotate(-5deg); }
    60% { -webkit-transform:    rotate(0deg); }
    70% { -webkit-transform:    rotate(-5deg); }
    80% { -webkit-transform:    rotate(5deg); }
    90% { -webkit-transform:    rotate(0deg); }
    100% { -webkit-transform:   rotate(-5deg); }
}

.item-shake {
    z-index: 10000;
    width: 125%;
    height: 125%;
    -webkit-animation-name: shake;
    -webkit-animation-duration: 0.5s;
    /*-webkit-transform-origin:50% 50%;*/
    -webkit-animation-iteration-count: infinite;
    -webkit-animation-timing-function: linear;
}
