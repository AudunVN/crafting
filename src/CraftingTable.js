import React, { Component } from 'react'
import { Panel } from 'react-bootstrap'

class CraftingTable extends Component {
  render () {
    return (
      <Panel header='Crafting Table'>
        <div className='crafting-holder'>
          <div className='crafting clearfix'>
            <div className='recipe'>
              <h6>Crafting</h6>
              <div id='crafting-table'>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
                <span className='grid grid-crafting'/>
              </div>
            </div>
            <div className='arrow'/>
            <div id='crafting-table-output'><span className='grid-large output'/></div>
          </div>
        </div>
      </Panel>
    )
  }
}

export default CraftingTable
