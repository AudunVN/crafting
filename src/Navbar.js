import React from 'react'
import { Navbar } from 'react-bootstrap'

const Nav = (props) =>
  <Navbar inverse collapseOnSelect>
    <Navbar.Header>
      <Navbar.Brand>
        <a href='#'>Crafting</a>
      </Navbar.Brand>
      <Navbar.Toggle />
    </Navbar.Header>
  </Navbar>

export default Nav
