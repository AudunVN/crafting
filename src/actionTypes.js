// Data
export const SET_CRAFTING_SLOT = 'SET_CRAFTING_SLOT'
export const RESET_CRAFTING_SLOT = 'RESET_CRAFTING_SLOT'
export const SET_FIRST_EMPTY_CRAFTING_SLOT = 'SET_FIRST_EMPTY_CRAFTING_SLOT'
export const SET_OUTPUT_SLOT = 'SET_OUTPUT_SLOT'
export const RESET_OUTPUT_SLOT = ''
// Options
export const SET_SHAPE = 'SET_SHAPE'
export const SET_EMPTY_SPACE = 'SET_EMPTY_SPACE'
export const SET_OUTPUT_RECIPE = 'SET_OUTPUT_RECIPE'
// Private
export const TOGGLE_SHOWING_CONTEXT_MENU = 'TOGGLE_SHOWING_CONTEXT_MENU'
export const TOGGLE_SHOWING_COUNT_MENU = 'TOGGLE_SHOWING_COUNT_MENU'