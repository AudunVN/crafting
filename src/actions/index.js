// Data reducer
export const SET_CRAFTING_SLOT = 'SET_CRAFTING_SLOT'
export const setCraftingSlot = (index, ingredient) => ({
  type: SET_CRAFTING_SLOT,
  payload: {
    index,
    ingredient
  }
})

export const RESET_CRAFTING_SLOT = 'RESET_CRAFTING_SLOT'
export const resetCraftingSlot = (index) => ({
  type: RESET_CRAFTING_SLOT,
  payload: {
    index
  }
})

export const SET_FIRST_EMPTY_CRAFTING_SLOT = 'SET_FIRST_EMPTY_CRAFTING_SLOT'
export const setFirstEmptyCraftingSlot = (ingredient) => ({
  type: SET_FIRST_EMPTY_CRAFTING_SLOT,
  payload: {
    ingredient
  }
})

export const SET_OUTPUT_SLOT = 'SET_OUTPUT_SLOT'
export const setOutputSlot = (ingredient) => ({
  type: SET_OUTPUT_SLOT,
  payload: {
    ingredient
  }
})

export const RESET_OUTPUT_SLOT = 'RESET_OUTPUT_SLOT'
export const resetOutputSlot = () => ({
  type: RESET_OUTPUT_SLOT
})

export const SET_GROUP = 'SET_GROUP'
export const setGroup = (group) => ({
  type: SET_GROUP,
  payload: group
})

// Options reducer
export const SET_SHAPE = 'SET_SHAPE'
export const setShape = (shape) => ({
  type: SET_SHAPE,
  payload: shape
})

export const SET_EMPTY_SPACE = 'SET_EMPTY_SPACE'
export const setEmptySpace = (emptySpace) => ({
  type: SET_EMPTY_SPACE,
  payload: emptySpace
})

export const SET_OUTPUT_RECIPE = 'SET_OUTPUT_RECIPE'
export const setOutputRecipe = (recipe) => ({
  type: SET_OUTPUT_RECIPE,
  payload: recipe
})

// Private reducer
export const TOGGLE_SHOWING_CONTEXT_MENU = 'TOGGLE_SHOWING_CONTEXT_MENU'
export const toggleContextMenu = () => ({
  type: TOGGLE_SHOWING_CONTEXT_MENU
})

export const TOGGLE_SHOWING_COUNT_MENU = 'TOGGLE_SHOWING_COUNT_MENU'
export const toggleCountMenu = () => ({
  type: TOGGLE_SHOWING_COUNT_MENU
})
