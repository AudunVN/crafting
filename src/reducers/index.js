import { combineReducers } from 'redux'

import Data from './Data'
import Options from './Options'

export default combineReducers({
  Data,
  Options
})