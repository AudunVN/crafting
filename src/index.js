import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

import reducer from './reducers'
import { applyMiddleware, createStore } from 'redux'
import { Provider } from 'react-redux'

let middleware = []

if (process.env.NODE_ENV === 'development') {
  // only add logger in development
  const { logger } = require('redux-logger')

  middleware.push(logger)
}

const store = createStore(
  reducer,
  applyMiddleware(...middleware)
)

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
