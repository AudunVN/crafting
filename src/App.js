import React  from 'react'
import { Col, Row } from 'react-bootstrap'
import { DragDropContext } from 'react-dnd'
import TouchBackend from 'react-dnd-touch-backend'

import Navbar from './components/Navbar'
import CraftingTable from './components/CraftingTable'
import Ingredients from './components/Ingredients'

import IngredientDragLayer from './components/IngredientDragLayer'

import 'bootstrap/dist/css/bootstrap.css'
import './App.css'

import './assets/arrow.png'
import './assets/Minecraft.woff'

const App = () =>
  <div className="container">
    <Navbar />
    <IngredientDragLayer />
    <Row>
      <Col md={6}>
        <CraftingTable />
      </Col>
      <Col md={6}>
        <Ingredients />
      </Col>
    </Row>
  </div>

export default DragDropContext(TouchBackend({ enableMouseEvents: true }))(App)
