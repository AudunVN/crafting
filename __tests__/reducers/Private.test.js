import reducer from '../../src/reducers/Private'

import { toggleContextMenu, toggleCountMenu } from '../../src/actions'

describe('Private Reducer', () => {
  it('should have a default state', () => {
    expect(
      reducer(undefined, {})
    ).toMatchSnapshot()
  })

  it('should toggle count menu', () => {
    expect(
      reducer(undefined, toggleCountMenu())
    ).toMatchSnapshot()
  })

  it('should toggle context menu', () => {
    expect(
      reducer(undefined, toggleContextMenu())
    ).toMatchSnapshot()
  })
})
