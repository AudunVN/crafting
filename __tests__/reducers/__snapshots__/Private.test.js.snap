exports[`Private Reducer should have a default state 1`] = `
Object {
  "showingContextMenu": false,
  "showingCountModal": false,
}
`;

exports[`Private Reducer should toggle context menu 1`] = `
Object {
  "showingContextMenu": true,
  "showingCountModal": false,
}
`;

exports[`Private Reducer should toggle count menu 1`] = `
Object {
  "showingContextMenu": false,
  "showingCountModal": true,
}
`;
